tests/test_flask.py

```diff
@@ -79,6 +80,52 @@ def teardown():
             yield client
             teardown()
 
+def test_failed_login(client):
+    #Testing the failed login page
+    with pytest.raises(InvalidCharactersException):
+        returned_value = client.post('/login', data=dict(
+            username="testTrainee#%#^",
+            password="password#^#$#"
+        ), follow_redirects=True)
+        assert returned_value.status_code == 200
+
+def test_failed_signup(client):
+    #Testing the failed signup page
+    with pytest.raises(InvalidCharactersException):
+        returned_value = client.post('/signup', data=dict(
+            username="testTrainee^#$^%^",
+            password="password)^^&&",
+            name="test^##%^",
+            repassword="password)^^&&",
+            location="&@%&&*",
+            phone="asdasdasd",
+            usertype="123456"
+        ), follow_redirects=True)
+        assert returned_value.status_code == 200
```

Comment:
Same as above, make independent `with pytest.raises()` blocks for each argument since right now the username if the statement is raising the exception.